packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -348,18 +352,18 @@ class CupertinoPageTransition extends StatelessWidget {
        _primaryPositionAnimation = (linearTransition ? primaryRouteAnimation :
          CurvedAnimation(
            parent: primaryRouteAnimation,
-           curve: Curves.easeOut,
-           reverseCurve: Curves.easeIn,
+           curve: Curves.linearToEaseOut,
```

Comment:
ditto

Reply:
To be honest, without a rigorous check comparing frames between native iOS and our Cupertino implementation, it's hard to say for sure which curve is the most accurate. Looking at a recording of native iOS frames, it seemed to me that a linear to ease curve was the most accurate. However, we are planning on making a framework to do native vs flutter testing in the future. 

Also, if you are testing this code, make sure you test in release mode as the animations feel quite different than they do in debug mode.

Reply:
With a bit too much time on my hands :smile: , I [plotted the frames](https://plot.ly/~marcoms/1) of a screen recording I made on my iPhone. The raw data is the following (measured with a ruler and lerp'd):

```
0
0.121212
0.208588
0.339393
0.442424
0.533333
0.618181
0.690909
0.751515
0.842424
0.878787
0.903030
0.921212
0.939393
0.951515
0.963636
0.969696
0.975757
0.981818
0.987878
0.990909
0.992424
0.993939
0.996969
1
```

Actually you're right that it is more of a linear curve followed by an ease out (and 400ms is about right too). The only difference is the native curve is fairly steeper.

Here it is modeled on cubic-bezier.com: http://cubic-bezier.com/#.35,.91,.33,.97

Hopefully plugging that into `Cubic()` will get us feeling more native. My only concern is whether Apple internally uses a parameterized curve that changes depending on the UI element :/

Reply:
With a bit too much time on my hands :smile: , I [plotted the frames](https://plot.ly/~marcoms/1) of a screen recording I made on my iPhone. The raw data is the following (measured with a ruler and lerp'd):

```
0
0.121212
0.208588
0.339393
0.442424
0.533333
0.618181
0.690909
0.751515
0.842424
0.878787
0.903030
0.921212
0.939393
0.951515
0.963636
0.969696
0.975757
0.981818
0.987878
0.990909
0.992424
0.993939
0.996969
1
```

Actually you're right that it is more of a linear curve followed by an ease out (and 400ms is about right too). The only difference is the native curve is fairly steeper.

Here it is modeled on cubic-bezier.com: http://cubic-bezier.com/#.35,.91,.33,.97

Hopefully plugging that into `Cubic()` will get us feeling more native. My only concern is whether Apple internally uses a parameterized curve that changes depending on the UI element :/

Reply:
Wow! That's awesome! Thanks for doing that! I'll add that function and see how it feels!

And by eyeball estimation, it looked like all their transition curves are the same across all their apps, tho it's hard to say absolutely for sure without checking. What I did notice tho is that the curve they use for animating a page back or forward if you release mid drag is different than the animation they use for when you tap a button to show a page.

Again, thanks for verifying this ! 😊